pkg/client/unversioned/request.go

```diff
@@ -585,8 +592,20 @@ func (r *Request) Watch() (watch.Interface, error) {
 	if client == nil {
 		client = http.DefaultClient
 	}
+	if r.backoffMgr == nil {
+		panic("nil backoffMgr should never occur")
```

Comment:
Your right: but I did feel like an early explicit panic would be better than a deep nested one....   Maybe thats not the right idiom to go with ?